File: samples/JavaScript/nth-root.js
function nthRoot(num, nArg, precArg) {
  var n = nArg || 2;
  var prec = precArg || 12;

  var x = 1; // Initial guess.
  for (var i=0; i<prec; i++) {
    x = 1/n * ((n-1)*x + (num / Math.pow(x, n-1)));
  }

  return x;
}

File: samples/Scala/nth-root.scala
object NthRoot {

  def main(args: Array[String]) {
    println(nthroot(3, 32))
  }

  def nthroot1(n: Int, a: Double): Double = {
    def loop(x0: Double) : Double = {
      val x1 = (1.0d/n * ((n - 1) * x0 + a/math.pow(x0, n-1)))
      if (x0 <= x1) x0
      else loop(x1)
    }

    return loop(a/2)
  }
}
